extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelRoutesBuilderClassBuildItem.java

```diff
@@ -17,20 +17,23 @@
 package org.apache.camel.quarkus.core.deployment;
 
 import io.quarkus.builder.item.MultiBuildItem;
-import io.quarkus.runtime.RuntimeValue;
 import org.apache.camel.RoutesBuilder;
+import org.jboss.jandex.DotName;
 
 /**
- * Holds the {@link RoutesBuilder} {@link RuntimeValue}.
+ * A {@link MultiBuildItem} holding class names of all {@link RoutesBuilder} implementations found via Jandex.
  */
-public final class CamelRoutesBuilderBuildItem extends MultiBuildItem {
-    private final RuntimeValue<RoutesBuilder> routesBuilder;
+public final class CamelRoutesBuilderClassBuildItem extends MultiBuildItem {
 
-    public CamelRoutesBuilderBuildItem(RuntimeValue<RoutesBuilder> routesBuilder) {
-        this.routesBuilder = routesBuilder;
-    }
+    private final DotName dotName;
 
-    public RuntimeValue<RoutesBuilder> getInstance() {
-        return routesBuilder;
+    public CamelRoutesBuilderClassBuildItem(DotName dotName) {
+        super();
```

Comment:
isn't this superfluous ?

Reply:
Eclipse seems to be doing that by default.